dps/spark/jobs/minhash_dedup_data.py

```diff
@@ -0,0 +1,67 @@
+import random
+from itertools import combinations
+
+from pyspark import SparkContext
+from pyspark.rdd import RDD
+
+from ..spark_session import spark_session
+from ..utils.io import read_line, to_json
+from ..utils.minhash_dedup import (shingle_word,
+                                   generate_minhash,
+                                   jaccard_by_hashvalues)
+
+
+def expand_instances_by_minhash(data, expand_size: int, n_gram: int, 
+                                seed: int = 1, char_level: bool = False):
+    shingles = shingle_word(data['text'], n_gram=n_gram, char_level=char_level)
+    minhashes = generate_minhash(shingles, num_perm=expand_size, seed=seed)
+
+    for mh in minhashes.tolist():
+        yield (str(mh), [dict(**data, shingles=shingles, hashvalues=minhashes)])
+
+
+def explore_dedup_instance(hash_groups, threshold: float = 0.8):
+    if len(hash_groups) <= 1:
+       return
+
+    group_represent_text = hash_groups[0]['text'] # not to remove all text instances in group.
+    pairs = combinations(hash_groups, 2)
+
+    for d_1, d_2 in pairs:
+        sim_score = jaccard_by_hashvalues(d_1['hashvalues'], d_2['hashvalues'])
+        if sim_score >= threshold:
+            dedup_text = [d_1['text'], d_2['text']]
+            if group_represent_text in dedup_text:
+                yield dedup_text[0] if dedup_text[0] != group_represent_text else dedup_text[1]
+            else:
+                yield random.choice(dedup_text)
+
+
+def minhash_deduplication(input_dir: str, output_dir: str, 
+                          n_dist: int = 10, n_output: int = 10, 
+                          num_expand: int = 10, sim_threshold: float = 0.8, 
+                          n_gram: int = 15, char_level: bool = False, seed: int = 1):
```

Comment:
how about using 13 as default? GPT3 used 13.

Reply:
Okay, I'll set 13 to `n_gram` as a default value :)
And, this parameter can be injected by command, so anyone can change this parameter setting.